Include/Intel/IndustryStandard/ProcessorInfo.h

```diff
@@ -128,16 +128,64 @@
 #define MSR_DRAM_PERF_STATUS    0x61B
 #define MSR_DRAM_POWER_INFO     0x61C
 
-/// x86 Page Address Translation
-enum {
-  PageAddressTranslationUncached       = 0,
-  PageAddressTranslationWriteCombining = 1,
-  PageAddressTranslationWriteThrough   = 4,
-  PageAddressTranslationWriteProtected = 5,
-  PageAddressTranslationWriteBack      = 6,
-  /// Uncached, but can be overriden by MTRR
-  PageAddressTranslationOverridableUncached = 7,
-};
+//
+// Page Attribute Table (PAT) cache types.
+// Note that the final memory cache type is defined according to the combination (ref. 11.5.2.2) of
+// PAT (ref. 11.12) and MTRR (ref. 11.11) and cannot be derived from either separately.
+// REF: Intel® 64 and IA-32 Architectures Software Developer’s Manual - Volume 3A: System Programming Guide, Part 1
+//
+typedef enum {
+  PatUncacheable    = 0,
+  PatWriteCombining = 1, ///< Special hardware burst mode (not L1-L3) intended for graphics memory (ref. 11.3.1).
+  PatWriteThrough   = 4,
+  PatWriteProtected = 5,
+  PatWriteBack      = 6,
+  PatUncached       = 7,
+  PatEntryMask      = 255
+} PAT_MEMORY_CACHE_TYPE;
+
+#define PAT_UNCACHEABLE      0
+#define PAT_WRITE_COMBINING  1
+#define PAT_WRITE_THROUGH    4
+#define PAT_WRITE_PROTECTED  5
+#define PAT_WRITE_BACK       6
+#define PAT_UNCACHED         7
+#define PAT_ENTRY_BIT_SIZE   8
+#define PAT_ENTRY_BIT_MASK   ((1 << PAT_ENTRY_BIT_SIZE) - 1)
```

Comment:
I was just copying the approach from here: https://github.com/acidanthera/audk/blob/master/UefiCpuPkg/Include/Library/MtrrLib.h#L81-L98

Thinking about it, the defines there are presumably present for this code: https://github.com/acidanthera/audk/blob/master/UefiCpuPkg/CpuDxe/CpuDxe.c#L535-L554 - i.e. basically if we ever hold the value in UINT8 and still want to check it.

Reply:
You can use enum values just fine for the last part, this is just another efk2 awkwardness.

Reply:
Given the above, since I don't currently do this but might, should I keep them?

(I think I should switch out from defines to enum values in PAT_DEFAULTS in any case.)